File: AoC2018/JavaScript/day1.js
//puzzle setup
const fs = require('fs');
var input = fs.readFileSync('./day1in.txt', 'utf8');
var arr = input.split(',');

var sum = 0;
var nums = [0];
//part 2?
//var ans = null;

for (i in arr)
{
    switch (arr[i].charAt(0))
    {
        case "+":
            sum += Number(arr[i].substr(1));
            pushCheck(sum);
            break;
        case "-":
            sum -= Number(arr[i].substr(1));
            pushCheck(sum);
            break;
    }
    //part 2?
    //if (ans) break;
}
console.log(nums[nums.length-1]);

function pushCheck(n)
{
    nums.push(n);
    //part 2?
    //if (nums.indexOf(n) != Number(i)+1) ans = n;
}

File: AoC2020/Python3/day1.py
#puzzle setup
with open('./day1in.txt','r') as pInput:
  pInput = pInput.read().splitlines()

#do stuff
pInput = [int(x) for x in pInput]
pInput.sort()

l = 0
r = len(pInput) - 1

while l < r:
    if pInput[l] + pInput[r] == 2020:
        print(pInput[l] * pInput[r])
        break
    elif pInput[l] + pInput[r] < 2020:
        l += 1
    else:
        r -= 1